File: VisitBosnia/VisitBosnia.Model/TouristFacility.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VisitBosnia.Model
{
    public class TouristFacility
    {

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public int CityId { get; set; }
        public int CategoryId { get; set; }

        public virtual Category Category { get; set; } = null!;
        public virtual City City { get; set; } = null!;

      
    }
}

File: VisitBosnia/visit_bosnia_mobile/lib/model/tourist_facility.dart
import 'category.dart';
import 'city/city.dart';

class TouristFacility {
  int? id;
  String? name;
  String? description;
  int? cityId;
  int? categoryId;
  Category? category;
  City? city;

  TouristFacility(
      {this.id,
      this.name,
      this.description,
      this.cityId,
      this.categoryId,
      this.category,
      this.city});

  TouristFacility.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    description = json['description'];
    cityId = json['cityId'];
    categoryId = json['categoryId'];
    category = json['category'] != null
        ? new Category.fromJson(json['category'])
        : null;
    city = json['city'] != null ? new City.fromJson(json['city']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['description'] = this.description;
    data['cityId'] = this.cityId;
    data['categoryId'] = this.categoryId;
    if (this.category != null) {
      data['category'] = this.category!.toJson();
    }
    if (this.city != null) {
      data['city'] = this.city!.toJson();
    }
    return data;
  }
}
